### PR TITLE
Add `@custom-media` support

### PR DESCRIPTION
**Pull request by umpox**
Relevant discussion: https://github.com/sourcegraph/sourcegraph/pull/20647#discussion_r626564076

We can move away from our Bootstrap/Scss mixins and more towards to CSS media queries spec with this

### CODE REVIEW

**Review by valerybugakov — APPROVED** [review]
That's great! It looks like a flawless upgrade 🎉

File: client/branded/src/global-styles/breakpoints.scss [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-// `$grid-breakpoints` is copied from `~bootstrap/scss/variables` because we cannot import a single variable from SCSS file
-// and we don't want to expose all Bootstrap SCSS variables ('bootstrap/scss/variables') to our CSS modules.
-// If possible, prefer CSS variables to SCSS variables.
-$grid-breakpoints: (
-    xs: 0,
-    sm: 576px,
-    md: 768px,
-    lg: 992px,
-    xl: 1200px,
-);
```
Comment (eseliger) [inline client/branded/src/global-styles/breakpoints.scss:13]: Don't we still need this while we use bootstrap grid? `d-md-block` for example might use a different breakpoint now, if I'm not mistaken

Comment (umpox) [reply client/branded/src/global-styles/breakpoints.scss:13]: These are already defined in Bootstrap as defaults, so I don't think so, unless we see value in redeclaring them here

**Comment by umpox** [comment]
Getting some strange behavior on Chromatic that I can't replicate locally, trying to rebuild...

**Review by eseliger — APPROVED** [review]
Good find!

**Pull request merged**